src/lsp/ParameterHintManager.js

```diff
@@ -0,0 +1,278 @@
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var _ = brackets.getModule("thirdparty/lodash");
+
+    var Commands        = brackets.getModule("command/Commands"),
+        AppInit         = brackets.getModule("utils/AppInit"),
+        CommandManager  = brackets.getModule("command/CommandManager"),
+        KeyEvent        = brackets.getModule("utils/KeyEvent"),
+        Menus           = brackets.getModule("command/Menus"),
+        Strings         = brackets.getModule("strings"), 
+        Utils           = require("lsp/Utils");
+
+    
+    var clientList = {};
+
+    /** @const {string} Show Function Hint command ID */
+    var SHOW_PARAMETER_HINT_CMD_ID   = "showParameterHint", // string must MATCH string in native code (brackets_extensions)
+        PUSH_EXISTING_HINT           = true,
+        OVERWRITE_EXISTING_HINT      = false,
+        hintContainerHTML            = require("text!lsp/ParameterHintTemplate.html");
+
+    var $hintContainer,    // function hint container
+        $hintContent,      // function hint content holder
+        hintState = {},
+        session;           // current editor session, updated by main
+
+    // Constants
+    var POINTER_TOP_OFFSET          = 4,    // Size of margin + border of hint.
+        POSITION_BELOW_OFFSET       = 4;    // Amount to adjust to top position when the preview bubble is below the text
+
+    var handleCursorActivity;
+
+    /**
+     * Update the current session for use by the Function Hint Manager.
+     *
+     * @param {Session} value - current session.
+     */
+    function setSession(value) {
+        session = value;
+    }
+
+    /**
+     * Test if a function hint is being displayed.
+     *
+     * @return {boolean} - true if a function hint is being displayed, false
+     * otherwise.
+     */
+    function isHintDisplayed() {
+        return hintState.visible === true;
```

Comment:
```suggestion
        return hintState.visible;
```
should be enough ;-)